deploy/prod-node/nginx/nginx.conf

```diff
@@ -68,9 +80,37 @@ http {
 		ssl_certificate_key /etc/nginx/ssl/live/backend.prod.open-assistant.io/privkey.pem;
 
 		location / {
-		    proxy_set_header Host $host;
-		    proxy_set_header X-Real-IP $remote_addr;
-		    proxy_pass http://127.0.0.1:8280;
+			proxy_set_header Host $host;
+			proxy_set_header X-Real-IP $remote_addr;
+			proxy_pass http://127.0.0.1:8280;
+		}
+	}
+
+	map $http_upgrade $connection_upgrade {
+		default upgrade;
+		''      close;
+	}
+
+	server {
+		listen 443 ssl http2;
+
+		server_name inference.prod.open-assistant.io;
+
+		ssl_certificate /etc/nginx/ssl/live/inference.prod.open-assistant.io/fullchain.pem;
+		ssl_certificate_key /etc/nginx/ssl/live/inference.prod.open-assistant.io/privkey.pem;
+
+		location / {
+			proxy_set_header Host $host;
+			proxy_set_header X-Real-IP $remote_addr;
+
+			proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+			proxy_set_header X-Forwarded-Proto $scheme;
+			proxy_set_header Upgrade $http_upgrade;
+			proxy_set_header Connection $connection_upgrade;
+			proxy_redirect off;
+			proxy_buffering off;
+
+			proxy_pass http://127.0.0.1:8285;
```

Comment:
I normally used something like this instead of hard coding the IP address. Helped in load balancing and scaling

 - docker will automatically resolve this to the correct address
  because we use the same name as the service: e.g "inference-server" in the docker compose
    
`
upstream inference_servers {
   
    server inference-server:8285;
}
`

`
 proxy_pass inference_servers;
`

Reply:
doesn't work because nginx is running in docker-compose while the rest of the services are deployed via ansible

Reply:
oh ok, understood.